model/pdata/optional.go

```diff
@@ -0,0 +1,38 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pdata // import "go.opentelemetry.io/collector/model/pdata"
+
+import (
+	"go.opentelemetry.io/collector/model/internal/data"
+)
+
+// OptionalDouble is an alias of OTLP OptionalDouble data type.
+type OptionalDouble struct {
+	orig data.OptionalDouble
+}
+
+// NewOptionalDouble returns a new OptionalDouble from the given byte array.
+func NewOptionalDouble(value float64) OptionalDouble {
+	return OptionalDouble{orig: data.NewOptionalDouble(value)}
+}
+
+// IsEmpty returns true if id doesn't contain at least one non-zero byte.
+func (t OptionalDouble) IsEmpty() bool {
+	return t.orig.IsEmpty()
+}
```

Comment:
Would be good to be kind of in sync with generated protobuf code, is this how they call the func?

Reply:
The generated protos checks for `nil` values. You can see the generated protos here

https://github.com/codeboten/opentelemetry-collector/pull/10/commits/7811abd5b1cf0e69a148a875a1c56a32cf2eda15#diff-7276c207e936f474133fad60cb485d044f4a8cc3931300c5b82b34fa2a238abdR810-R821

Here's an example of GetMin:
```
func (x *HistogramDataPoint) GetMin() float64 {
	if x != nil && x.Min != nil {
		return *x.Min
	}
	return 0
}
```

It uses the following to marshal:

```
if m.Max != nil {
		i -= 8
		binary.LittleEndian.PutUint64(dAtA[i:], uint64(math.Float64bits(float64(*m.Max))))
		i--
		dAtA[i] = 0x61
	}
	if m.Min != nil {
		i -= 8
		binary.LittleEndian.PutUint64(dAtA[i:], uint64(math.Float64bits(float64(*m.Min))))
		i--
		dAtA[i] = 0x59
	}
```

Following the `IsEmpty` pattern was done to remain consistent with the trace/span ID custom types already present in pdata.